src/input/json_to_storage.cc

```diff
@@ -440,49 +825,45 @@ StorageBase * JSONToStorage::record_automatic_conversion(JSONPath &p, const Type
 
 
 
-StorageBase * JSONToStorage::make_storage(JSONPath &p, const Type::AbstractRecord *abstr_rec)
+StorageBase * JSONToStorage::make_storage(PathBase *p, const Type::AbstractRecord *abstr_rec)
 {
-    if (p.head()->type() == json_spirit::obj_type) {
+	bool has_desc_index;
+	try {
+		has_desc_index = p->has_descendent_index( abstr_rec->get_selection_default().has_value_at_declaration() );
+	} catch(ExcInputError &e) {
+		e << EI_InputType(abstr_rec->desc());
+		throw;
+	}
 
-        JSONPath type_path(p);
-        if ( type_path.down("TYPE") == NULL ) {
```

Comment:
replace this condition with call to get_descendant_name and test it against empty string.